backend/node/base.py

```diff
@@ -180,9 +180,54 @@ async def get_contract_data(
             from_address, calldata, readonly=True, is_init=False
         )
 
-    async def get_contract_schema(self, code: str) -> str:
+    async def _execution_finished(self, res: genvmbase.ExecutionResult):
+        msg_handler = self.msg_handler
+        if msg_handler is None:
+            return
+        msg_handler.send_message(
+            LogEvent(
+                name="execution_finished",
+                type=EventType.INFO,
+                scope=EventScope.GENVM,
+                message="execution finished",
+                data={
+                    "result": f"{res.result!r}",
+                    "stdout": res.stdout,
+                    "stderr": res.stderr,
+                    "genvm_log": res.genvm_log,
+                },
+            )
+        )
+
+    async def get_contract_schema(self, code: str) -> str | dict:
```

Comment:
we should add a better type here probably, but that is more related to dumping all errors in a more user friendly way in general I think... Because right now on exception only limited information is shown and it's arguments are not split into separate json fields but are rather encoded as a string, which leads to unreadable backslash madness in the ui